akka-stream/src/main/scala/akka/stream/javadsl/Source.scala

```diff
@@ -250,6 +250,11 @@ object Source {
    * The actor will be stopped when the stream is completed, failed or canceled from downstream,
    * i.e. you can watch it to get notified when that happens.
    *
+   * An async boundary is automatically created after the returned actorRef source, so downstream
+   * operations will not occur in the same thread as the Actor ingesting messages. If you select too
```

Comment:
Perhaps I was unclear. Yes, of course will never run anything on the sender's thread, that would not be acceptable. We have two async boundaries here, one between the sender and the actor, and one between the actor and other downstream processing. The second boundary is not the reason for that we are not running in the sender's thread. It is the first boundary that makes that happen.

I don't think users should have to care about these boundaries. They should care about that there is no backpressure to the sender so if they send faster than the stream consumes the buffer will overflow.